sim/hunter/items.go

```diff
@@ -569,6 +570,124 @@ func init() {
 		hunter.ItemSwap.RegisterProc(SignetOfTheUnseenPath, procAura)
 	})
 
+	core.NewItemEffect(StringsOfFate, func(a core.Agent) {
+		hunter := a.(HunterAgent).GetHunter()
+
+		hasMeleeSpecialist := hunter.HasRune(proto.HunterRune_RuneBeltMeleeSpecialist)
+
+		// After using the bow active you cannot gain strands for 30sec.
+		noGainAura := hunter.GetOrRegisterAura(core.Aura{
+			Label:    "Strand Of Fate - No Stack Gain",
+			ActionID: core.ActionID{SpellID: 1231603},
+			Duration: time.Second * 30,
+		})
+
+		// Tracks the number of strands of fate, up to 5 stacks if you're ranged or 4 if you're using the melee version.
+		// TODO: Ranged Hunter's version lets them consume these stacks to move while shooting
+		stacksAura := hunter.GetOrRegisterAura(core.Aura{
+			Label:     "Strand Of Fate - Stacks",
+			ActionID:  core.ActionID{SpellID: 1232946},
+			MaxStacks: core.TernaryInt32(hasMeleeSpecialist, 4, 5),
+			Duration:  time.Second * 20,
+			OnStacksChange: func(aura *core.Aura, sim *core.Simulation, oldStacks, newStacks int32) {
+				if hasMeleeSpecialist {
+					strengthChange := float64(20 * (newStacks - oldStacks))
+					hunter.Unit.AddStatDynamic(sim, stats.Strength, strengthChange)
+				}
+			},
+		})
```

Comment:
🔨 Rather than checking this within the function, we can just dynamically add the `OnStacksChange` to the `stacksAura` if they actually have melee specialist

```suggestion
		})
		
		if hasMeleeSpecialist {
		  stacksAura.OnStacksChange = func(aura *core.Aura, sim *core.Simulation, oldStacks, newStacks int32) {
				if hasMeleeSpecialist {
					strengthChange := float64(20 * (newStacks - oldStacks))
					hunter.Unit.AddStatDynamic(sim, stats.Strength, strengthChange)
				}
			},
		}
```